static/js/ui_init.js

```diff
@@ -94,10 +94,7 @@ function message_hover(message_row) {
     if (current_message_hover && rows.id(current_message_hover) === id) {
         return;
     }
-    // Don't allow on-hover editing for local-only messages
-    if (message_row.hasClass("local")) {
-        return;
-    }
+
     const message = current_msg_list.get(rows.id(message_row));
     message_unhover();
     current_message_hover = message_row;
```

Comment:
Maybe worth adding a comment next to the `is_topic_editable` check noting that locally echoed messages fit into that code path as well?

Reply:
Sure, this will be better to track down.